shell/edit/provisioning.cattle.io.cluster/rke2.vue

```diff
@@ -1672,7 +1732,36 @@ export default {
       } else {
         set(this.agentConfig, 'protect-kernel-defaults', false);
       }
-    }
+    },
+
+    /**
+     * Handle k8s changes side effects, like PSP and PSA resets
+     */
+    handleKubernetesChange(value) {
+      if (value) {
+        set(this.value.spec, 'defaultPodSecurityAdmissionConfigurationTemplateName', '');
+        set(this.value.spec, 'defaultPodSecurityPolicyTemplateName', '');
```

Comment:
We don't want to have cases where the PSP is not valid for the current version (1.25+) or if we set the PSA.
Since there were no better guidelines to match all the cases, I opted for the simplest and safest option.
Would you recommend to keep it if version is lower than 1.25?

Reply:
This has been agreed with @kwwii to be ok.